x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,174 @@
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      query: {
+        match_all: {},
+      },
+      size: 1,
+    });
+
+    return !!queryResult.hits.hits.length;
+  } catch (e) {
+    return false;
+  }
+};
+
+const getHealthyAgents = (enrichedAgentPolicies: GetAgentPoliciesResponseItem[]): number =>
+  enrichedAgentPolicies.reduce(
+    (previousValue, currentValue) => previousValue + (currentValue.agents || 0),
+    0
+  );
+
+const getMinutesPassedSinceMoment = (momentInput: MomentInput): number =>
+  moment().diff(moment(momentInput), 'minutes');
+
+const getStatus = (
+  findingsIndexExists: boolean,
+  installedIntegrations: number,
+  healthyAgents: number,
+  minutesPassedSinceInstallation: number
+): Status => {
+  if (findingsIndexExists) return 'indexed';
+  if (installedIntegrations === 0) return 'not-installed';
+  if (healthyAgents === 0) return 'not-deployed';
+  if (minutesPassedSinceInstallation <= INDEX_TIMEOUT_IN_MINUTES) return 'indexing';
+  if (minutesPassedSinceInstallation > INDEX_TIMEOUT_IN_MINUTES) return 'index-timeout';
+
+  throw new Error('could not determine csp setup status');
+};
+
+const getCspSetupStatus = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  packageService: PackageService,
+  packagePolicyService: PackagePolicyServiceInterface,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<CspSetupStatus> => {
+  const [findingsIndexExists, installationPackageInfo, latestPackageInfo, installedIntegrations] =
+    await Promise.all([
+      isFindingsIndexExists(esClient),
+      packageService.asInternalUser.getInstallation(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
+      packageService.asInternalUser.fetchFindLatestPackage(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
+      getCspPackagePolicies(soClient, packagePolicyService, CLOUD_SECURITY_POSTURE_PACKAGE_NAME, {
+        per_page: 10000,
+      }),
+    ]);
+
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    installedIntegrations.items,
+    agentPolicyService
+  );
+
+  const enrichedAgentPolicies = await addRunningAgentToAgentPolicy(
+    agentService,
+    agentPolicies || []
+  );
+  const healthyAgents = getHealthyAgents(enrichedAgentPolicies);
+  const installedIntegrationsTotal = installedIntegrations.total;
+  const latestPackageVersion = latestPackageInfo.version;
+
+  const status = getStatus(
+    findingsIndexExists,
+    installedIntegrationsTotal,
+    healthyAgents,
+    getMinutesPassedSinceMoment(installationPackageInfo?.install_started_at || 0)
+  );
+
+  if (status === 'not-installed')
+    return {
+      status,
+      latestPackageVersion,
+      healthyAgents,
+      installedIntegrations: installedIntegrationsTotal,
+    };
+
+  return {
+    status,
+    latestPackageVersion,
+    healthyAgents,
+    installedIntegrations: installedIntegrationsTotal,
+    installedPackageVersion: installationPackageInfo?.install_version,
```

Comment:
As mentioned, `installationPackageInfo` can return `undefined` for various reasons. for example, uninstalled package and failed API call.
another scenario is having `indexed` status (having findings == `indexed`) but integration is not installed. which is "mixed" type between `indexed` and `not-installed`. it's ok for now in order to simplify the API. but in the future, we might want to create a separate type for each possible status